tetris_js/js/GameField.js

```diff
@@ -0,0 +1,395 @@
+            if (!this.tiles.get(index).occupied) {return false;}
+        } return true;
+    }
+
+    /**
+     * Moves the 'occupied' status 'down' for all tiles above a given border tile (the start of a full row);
+     * @param {number} borderIndex - the start ('first ftrom the left') position of the full row;
+     */
+    moveAllOccupiedDown = borderIndex => {
+        for (let index = (borderIndex - 2); index >= 1; index--) {
+            if (this.tiles.get(index).occupied) {
+                this.tiles.get(index).occupied = false;
+                if (index + 11 < 220) {
+                    this.tiles.get(index + 11).occupied = true;
+                }
+            }
+        }
+    }
+
+    /**
+     * Removes the 'occupied' status from all the tiles in a given matrix row;
+     * @param {number} startingIndex - the first tile of a given row;
+     */
+    deoccupyRow = startingIndex => {
+        let start = startingIndex;
+        let stop = start + 10;
+
+        for (let index = start; index < stop; index++) {
+            this.tiles.get(index).occupied = false;
+        }
+    }
+
+    /**
+     * Checks whether it is possible to rotate the current shape ie. by using the invisible 11th column;
+     * @param {Array} coordinates - the coordinates of the currently active shape;
+     */
+    checkRotationAvailability = (coordinates) => {
+        for (const coordinate of coordinates) {
+            if (coordinate % 11 === 0 || coordinate > 220 || coordinate < 1 || this.tiles.get(coordinate).occupied && !this.tilesActive.includes(coordinate)) {
+                return false;
+            }
+        } return true;
+    }
+
+    /**
+     * Depending on the name of the current shape runs the reelevant rotation method adjusting the shape opsition;
+     */
+    rotateCurrentShape = () => {
+        switch (this.currentShape.name) {
+            case "TRIANGLE":
+                this.rotateTriangle();
+                break
+            case "BAR":
+                this.rotateBar();
+                break
+            case "STEP":
+                this.rotateStep();
+                break
+            case "EL":
+                this.rotateEl();
+        }
+    }
+
+    rotateTriangle = () => {
+        let activeTiles = this.tilesActive;
+        let tileOne = activeTiles[0];
+        let tileTwo = activeTiles[1];
+        let tileThree = activeTiles[3];
+
+        switch (this.currentShape.rotationState) {
+            case Angles.DEFAULT:
+                if (this.checkRotationAvailability([tileOne + 12, tileTwo - 10, tileThree + 10])) {
+                    this.deactivateTiles(this.tilesActive);
+                    activeTiles[0] += 12;
+                    activeTiles[1] -= 10;
+                    activeTiles[3] += 10;
+                    this.currentShape.rotationState = Angles.ONEROTATION;
+                    this.activateTiles(this.tilesActive);
+                    break
+                } else {break;}
```

Comment:
Proponuje zebys sie zapoznal z [eslintem](https://eslint.org/) to jeden ze standardowych linterow JSowych, tj. program do statycznego sprawdzania spojnosci / poprawnosci kodu. Sa to tego dobre extensiony dla VSCode, potrzebujesz tylko utworzyc config i najlepiej wybrac jedna z uznanych konwencji, czy tam configa `recommended`. Oczywiscie to wszystko jest bardzo uznaniowe, ale utworzone na podstawie doswiadczen spolecznosci developerskiej. Ale z mojego doswiadczenia, kazdy zespol/organizacja ma swoj zestaw regul
```suggestion
                };
                break;
```